drivers/abp2/abp2.c

```diff
@@ -0,0 +1,227 @@
+        return ABP2_ERRBUS;
+    }
+#if defined(MODULE_ABP2_SPI)
+    spi_acquire(dev->params->spi, dev->params->cs, SPI_MODE_0, dev->params->clk);
+    status = spi_transfer_byte(dev->params->spi, dev->params->cs, false, ABP2_CMD_NOP);
+    spi_release(dev->params->spi);
+#else
+#pragma message("implement I2C code here")
+#endif
+
+    return status;
+}
+
+int abp2_read_raw(const abp2_t *dev, abp2_raw_t *raw_values)
+{
+    uint8_t status = 0xFF;      /* sensor status byte */
+    int res = ABP2_OK;
+    int len = sizeof(data_tx_nop_read) / sizeof(data_tx_nop_read[0]);
+    uint8_t data_rx[ABP2_RX_BUF_LEN];
+
+#if defined(MODULE_ABP2_SPI)
+    res = spi_init_cs(dev->params->spi, dev->params->cs);
+#else
+#pragma message("implement I2C code here")
+#endif
+    if (res) {
+        return ABP2_ERRBUS;
+    }
+#if defined(MODULE_ABP2_SPI)
+    spi_acquire(dev->params->spi, dev->params->cs, SPI_MODE_0, dev->params->clk);
+    spi_transfer_bytes(dev->params->spi, dev->params->cs, false, data_tx_nop_read, data_rx, len);
+    spi_release(dev->params->spi);
+#else
+#pragma message("implement I2C code here")
+#endif
+    status = data_rx[0];
+    if (status & ABP2_STATUS_MASK) {
+        return ABP2_ERRSENS;
+    }
+    raw_values->cntPress = 0x00FFFFFF & (data_rx[1] << 16 | data_rx[2] << 8 | data_rx[3]);
+    raw_values->cntTemp = 0x00FFFFFF & (data_rx[4] << 16 | data_rx[5] << 8 | data_rx[6]);
+
+    return res;
+}
+
+int abp2_measure_read(const abp2_t *dev, abp2_raw_t *raw_values)
+{
+    uint8_t status = 0xFF;     /* sensor status byte */
+    int res = ABP2_OK;
+    int len = sizeof(data_tx_meas_read) / sizeof(data_tx_meas_read[0]);
+    uint8_t data_rx[ABP2_RX_BUF_LEN];
+
+#if defined(MODULE_ABP2_SPI)
+    res = spi_init_cs(dev->params->spi, dev->params->cs);
+#else
+#pragma message("implement I2C code here")
+#endif
+    if (res) {
+        return ABP2_ERRBUS;
+    }
+#if defined(MODULE_ABP2_SPI)
+    spi_acquire(dev->params->spi, dev->params->cs, SPI_MODE_0, dev->params->clk);
+    spi_transfer_bytes(dev->params->spi, dev->params->cs, false, data_tx_meas_read, data_rx, len);
+    spi_release(dev->params->spi);
+#else
+#pragma message("implement I2C code here")
+#endif
+    status = data_rx[0];
+    if (status & ABP2_STATUS_MASK) {
+        return ABP2_ERRSENS;
+    }
+    raw_values->cntPress = 0x00FFFFFF & (data_rx[1] << 16 | data_rx[2] << 8 | data_rx[3]);
+    raw_values->cntTemp = 0x00FFFFFF & (data_rx[4] << 16 | data_rx[5] << 8 | data_rx[6]);
+
+    return res;
+}
+
+float abp2_pressure(const abp2_t *dev, const abp2_raw_t *raw_values)
+{
+    float press = 0.0f;
+
+    press = (raw_values->cntPress - cntPMin);
+    press /= (cntPMax - cntPMin);
+    press *= (dev->params->rangeMax - dev->params->rangeMin);
+    press += dev->params->rangeMin - pOffset;
+    return press;
+}
+
+float abp2_temperature(const abp2_t *dev, const abp2_raw_t *raw_values)
+{
+    (void)(dev);        /* keep unused parameter to homogenize function prototypes */
```

Comment:
Why not call `abp2_measure_read()` here so the user only has to call a single function?

Reply:
`abp2_measure_read()` reads both P and T out of the sensor. If you do the same for the `abp2_pressure()`, it means 2 bus transfers instead of 1.

Reply:
Direct access functions to P and T now available.